### PR TITLE
[Doc]Add details about pipeline.workers

### PR DESCRIPTION
**Pull request by karenzone**
Addresses  #10930 - [Doc] Add more info on how number of pipeline.workers is determined

PREVIEW: http://logstash_11474.docs-preview.app.elstc.co/guide/en/logstash/master/logstash-settings-file.html

### CODE REVIEW

**Comment by karenzone** [comment]
FYI: @kunisen and @rwaight.  Does this meet the need from you and your customers?  Anything else we can do to improve the docs/user experience?

**Comment by rwaight** [comment]
This LGTM, but would want confirmation from @kunisen as well here.

**Comment by kunisen** [comment]
Thanks! @karenzone for working on this. 
LGTM!
Just FYI, this was raised because of Hyper-Threading will double the thread (e.g. core:12, thread:24).
By adding the ref link of this will make the description very clearly.
Thanks again!

File: docs/static/settings-file.asciidoc [hunk 1]
```diff
@@ -87,6 +87,7 @@ The `logstash.yml` file includes the following settings.
 
 | `pipeline.workers`
 | The number of workers that will, in parallel, execute the filter and output stages of the pipeline.
+  This setting uses the https://docs.oracle.com/javase/7/docs/api/java/lang/Runtime.html#availableProcessors()[`java.lang.Runtime.getRuntime.availableProcessors`] value.
```
Comment (jsvd) [inline docs/static/settings-file.asciidoc]:
```suggestion
  This setting uses the https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Runtime.html#availableProcessors()[`java.lang.Runtime.getRuntime.availableProcessors`] value.
```

**Comment by yaauie** [comment]
I have confirmed that that the number of available processors (as reported by [`java.lang.Runtime.getRuntime.availableProcessors`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Runtime.html#availableProcessors)) is used as a default for when a pipeline's `pipeline.workers` is not explicitly specified.

As with many other pipeline-specific settings, there are several places where it can be specified at different scopes, and the flow through to the default is:

 - the pipeline-specific explicit setting from the pipeline's definition (e.g., in `pipelines.yml` or in Centralized Pipeline Management); OR
 - the explicit global setting, which can be defined in `logstash.yml` and may be overridden by a command-line flag; OR
 - the default value for the setting.

In this case, to override `pipeline.workers` at the command-line, one would use `--pipeline-workers X` or the shorthand `-w X`.

**Comment by karenzone** [comment]
@yaauie Thanks for the additional info.  I tried to keep the description as mainstream, yet accurate, as possible.  Sometimes too much detail can detract from the main message.  Do I need to include more info on ways to override the setting?

**Pull request closed without merging**